backend/app/Http/Controllers/FollowingController.php

```diff
@@ -0,0 +1,119 @@
+
+use Illuminate\Http\Request;
+use App\Models\ViewModels\Response;
+use App\Models\UserRelationshipModel;
+use App\Models\UserActivitiesModel;
+use DB;
+use Auth;
+use App\Models\User;
+
+class FollowingController extends Controller
+{
+    private $response;
+    public function __construct() {
+        $this->response = new Response();
+    }
+
+    public function get_user_activities(){
+        $usra = UserActivitiesModel::with('user_relationship.following','user_relationship.followed_back', 'user')->get();
+        $this->response->status_code = 1;
+        $this->response->message = "success";
+        $this->response->data = $usra;
+
+        return response()->json($this->response);
+    }
+
+    public function get_user_relationship(){
+        $usr = UserRelationshipModel::all();
+        $this->response->status_code = 1;
+        $this->response->message = "success";
+        $this->response->data = $usr;
+
+        return response()->json($this->response);
+    }
+
+    public function follow_user(Request $request){
+        try{
+            DB::beginTransaction();
+            
+            $my_id = Auth::user()->user_id;
+
+            $user_relationship = new UserRelationshipModel();
+            $user_relationship->following_id = $request->following_id;
+            $user_relationship->save();
+
+            $user_activities = new UserActivitiesModel();
+            $user_activities->user_relationship_id = $user_relationship->user_relationship_id;
+            $user_activities->user_id = Auth::user()->user_id;
+            $user_activities->save();
+
+            $this->response->status_code = 1;
+            $this->response->message = "success";
+            $this->response->data = "Followed Successfuly!";
+            
+            DB::commit();
+            return response()->json($this->response);
+        } catch (\Exception $ex) {
+            DB::rollback();
+            $this->response->status_code = 0;
+            $this->response->message = "error";
+            $this->response->data = "Data Error";
+
+            return response()->json($this->response);
+        }
+    }
+
+    public function follow_back(Request $request){
+        try {
+            DB::beginTransaction();
+
+            $my_id = Auth::user()->user_id;
+            $usra = UserActivitiesModel::with(['user_relationship' => function($q) use($my_id) {
+                $q->where('following_id', $my_id);
+            }])
+            ->where('user_id', $request->follow_back_id)
+            ->first();
+
+            $followed_back = UserRelationshipModel::find($usra->user_relationship_id);
+            $followed_back->followed_id = $my_id;
+            $followed_back->save();
+
+            $following = new UserRelationshipModel();
+            $following->following_id = $request->follow_back_id;
+            $following->followed_id = $request->follow_back_id;
+            $following->save();
+
+            $user_activities = new UserActivitiesModel();
```

Comment:
This is not included in our scope.
So please remove code related to UserActivity